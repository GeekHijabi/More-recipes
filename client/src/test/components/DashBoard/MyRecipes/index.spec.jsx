import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ConnectedRecipeAdmin, { RecipeAdmin }
  from '../../../../js/components/DashBoard/MyRecipes';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const props = {
  apiLoginUser: jest.fn(() => Promise.resolve({ ok: true })),
  apiCreateRecipe: jest.fn(() => Promise.resolve({ ok: true })),
  apiGetMyRecipe: jest.fn(() => Promise.resolve({ ok: true })),
  apiDeleteRecipe: jest.fn(() => Promise.resolve({ ok: true })),
  apiEditRecipe: jest.fn(() => Promise.resolve({ ok: true })),
  onViewRecipe: jest.fn(() => Promise.resolve({ ok: true })),
  myRecipes: [
    {
      id: 1,
      recipeName: 'rice',
      description: 'boil rice',
      ingredients: 'rice, water'
    },
    {
      id: 2,
      recipeName: 'beans',
      description: 'cook beans',
      ingredients: 'beans, water'
    }
  ],
  errorMessage: 'error',
  pageCount: 122,
  isLoadingRecipe: false,
  history: {
    push: () => Promise.resolve({ ok: true }),
  }

};


describe('RecipeAdmin snapshot', () => {
  it('should render correctly', () => {
    const recipeAdmin = shallow(<RecipeAdmin {...props} />);
    expect(recipeAdmin).toMatchSnapshot();
  });
});

describe('RecipeAdmin component', () => {
  it('should recieve next props', () => {
    const spy = sinon.spy(RecipeAdmin.prototype, 'componentWillReceiveProps');
    const nextProps = {
      apiLoginUser: jest.fn(() => Promise.resolve({ ok: true })),
      apiCreateRecipe: jest.fn(() => Promise.resolve({ ok: true })),
      apiGetMyRecipe: jest.fn(() => Promise.resolve({ ok: true })),
      apiDeleteRecipe: jest.fn(() => Promise.resolve({ ok: true })),
      apiEditRecipe: jest.fn(() => Promise.resolve({ ok: true })),
      onViewRecipe: jest.fn(() => Promise.resolve({ ok: true })),
      myRecipes: [
        {
          id: 1,
          recipeName: 'rice',
          description: 'boil rice',
          ingredients: 'rice, water'
        },
        {
          id: 2,
          recipeName: 'beans',
          description: 'cook beans',
          ingredients: 'beans, water'
        }
      ],
      errorMessage: 'error',
      pageCount: 122,
      page: 2,
      isLoadingRecipe: false

    };
    const wrapper = shallow(<RecipeAdmin {...nextProps} />);
    wrapper.setProps({ prop: 12 });
    expect(spy.calledOnce).toEqual(true);
  });
  describe('onPageChange', () => {
    it('should change the page', () => {
      const current = {
        selected: 2
      };
      const wrapper = shallow(<RecipeAdmin {...props} />);
      wrapper.instance().onPageChange(current);
      expect(wrapper.instance().props.apiGetMyRecipe).toBeCalled();
    });
  });

  describe('onDelete', () => {
    it('should delete the recipe', () => {
      const wrapper = shallow(<RecipeAdmin {...props} />);
      wrapper.instance().onDelete(2);
      expect(wrapper.instance().props.apiDeleteRecipe).toBeCalled();
      expect(wrapper.instance().props.apiDeleteRecipe.mock.calls.length).toEqual(1);
    });
  });

  describe('toggle', () => {
    it('should toggle the modal', () => {
      const wrapper = shallow(<RecipeAdmin {...props} />);
      wrapper.setState({ modal: true });
      wrapper.instance().toggle();
      expect(wrapper.state().modal).toEqual(false);
    });
  });
  describe('viewRecipe', () => {
    it('should display recipe information', () => {
      const wrapper = shallow(<RecipeAdmin {...props} />);
      wrapper.instance().viewRecipe(2);
      expect(wrapper.instance().props.onViewRecipe).toBeCalled();
      expect(wrapper.instance().props.onViewRecipe.mock.calls.length)
        .toEqual(1);
    });
  });

  describe('Connected RecipeAdmin component', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        recipe: {
          allRecipes: {},
          myRecipes: [
            {

            }
          ],
          errorMessage: ''

        },
      });
      const wrapper = shallow(<ConnectedRecipeAdmin store={store} />);
      expect(wrapper.length).toBe(1);
    });
  });
});
