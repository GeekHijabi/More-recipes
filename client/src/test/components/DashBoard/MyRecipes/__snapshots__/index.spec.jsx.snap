// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeAdmin snapshot should render correctly 1`] = `
<div>
  <withRouter(Connect(RecipeHeader)) />
  <div
    style={
      Object {
        "margin": "0 auto",
        "minHeight": "795px",
        "width": "85%",
      }
    }
  >
    <section
      className="row"
      id="list"
    >
      <div
        className="col-xs-12 col-sm-12 col-md-12 col-lg-12 heading"
      >
        <h4
          className="mb-5"
        >
          <span>
            My Recipe
          </span>
        </h4>
        <i
          className="fa fa-plus-circle fa-2x fa-icons"
          data-toggle="tooltip"
          onClick={[Function]}
          role="button"
          tabIndex="-1"
          title="Add recipe"
        />
      </div>
      <AddRecipeModal
        createRecipe={[MockFunction]}
        errorMessage="error"
        isLoadingRecipe={true}
        isOpen={false}
        toggle={[Function]}
      />
      <div
        className="container-fluid"
      >
        <div
          className="row"
        >
          <AdminCardItem
            deleteRecipe={[Function]}
            editRecipe={[MockFunction]}
            key="1"
            onViewRecipe={[Function]}
            recipe={
              Object {
                "description": "boil rice",
                "id": 1,
                "ingredients": "rice, water",
                "recipeName": "rice",
              }
            }
          />
          <AdminCardItem
            deleteRecipe={[Function]}
            editRecipe={[MockFunction]}
            key="2"
            onViewRecipe={[Function]}
            recipe={
              Object {
                "description": "cook beans",
                "id": 2,
                "ingredients": "beans, water",
                "recipeName": "beans",
              }
            }
          />
          <div
            className="col-xs-12 col-sm-12 col-md-12 col-lg-12"
          />
        </div>
      </div>
    </section>
  </div>
  <PaginationBoxView
    activeClassName="page-item active"
    breakClassName="text-center"
    breakLabel="..."
    containerClassName="container pagination justify-content-center"
    disableInitialCallback={false}
    disabledClassName="disabled"
    marginPagesDisplayed={3}
    nextClassName="page-item"
    nextLabel="Next"
    nextLinkClassName="page-link"
    onPageChange={[Function]}
    pageClassName="page-item"
    pageCount={1}
    pageLinkClassName="page-link"
    pageRangeDisplayed={5}
    previousClassName="page-item"
    previousLabel="Previous"
    previousLinkClassName="page-link"
  />
  <Footer />
</div>
`;
